modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/marshaller/PartitionCommandsMarshallerImpl.java

```diff
@@ -43,6 +43,8 @@ protected void beforeWriteMessage(Object o, ByteBuffer buffer) {
 
     @Override
     public <T> T unmarshall(ByteBuffer raw) {
+        raw = raw.duplicate();
```

Comment:
Why do we have two identical classes?

Reply:
It's a temporary solution while we're migrating to zone-based affinity